corehq/apps/cowin/repeaters.py

```diff
@@ -0,0 +1,73 @@
+from django.utils.translation import ugettext_lazy as _
+
+from corehq import toggles
+from corehq.apps.cowin.repeater_generators import (
+    BeneficiaryRegistrationPayloadGenerator,
+    BeneficiaryVaccinationPayloadGenerator,
+)
+from corehq.apps.hqcase.utils import update_case
+from corehq.motech.repeaters.models import CaseRepeater
+
+
+class BaseCOWINRepeater(CaseRepeater):
+    class Meta:
+        app_label = 'repeaters'
+
+    @classmethod
+    def available_for_domain(cls, domain):
+        return toggles.COWIN_INTEGRATION.enabled(domain)
+
+    def get_headers(self, repeat_record):
+        headers = super(BaseCOWINRepeater, self).get_headers(repeat_record)
+        headers.update({
+            'Accept-Language': 'en_US',
+            'User-Agent': '',
+            'Accept': 'application/json',
+            'Content-Type': 'application/json',
+        })
+        return headers
+
+
+class BeneficiaryRegistrationRepeater(BaseCOWINRepeater):
+    payload_generator_classes = (BeneficiaryRegistrationPayloadGenerator,)
+    friendly_name = _("Register beneficiaries on COWIN")
+
+    def get_headers(self, repeat_record):
+        headers = super(BeneficiaryRegistrationRepeater, self).get_headers(repeat_record)
+        headers.update({
+            'X-Api-Key': self.connection_settings.plaintext_password,
+        })
+        return headers
+
+    def handle_response(self, response, repeat_record):
+        attempt = super().handle_response(response, repeat_record)
+        if response.status_code == 200:
+            beneficiary_reference_id = response.json()['beneficiary_reference_id']
+            update_case(self.domain, repeat_record.payload_id, case_properties={
+                'cowin_id': beneficiary_reference_id
+            })
```

Comment:
It could be useful to use the `xmlns` or `device_id` params to indicate that this change came from `BeneficiaryRegistrationRepeater`. We do this for OpenMRS, DHIS2 and FHIR repeaters (including the repeater ID) in order to have an audit trail.